src/CMakeLists.txt

```diff
@@ -450,6 +453,9 @@ install(
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libopenshot
   FILES_MATCHING PATTERN "*.h"
 )
+install(FILES ${ProtobufHeaders}
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libopenshot
+)
```

Comment:
Currently, because these are `#include`d in the effect headers, I have the build installing these... but, it would be _really_ nice if we **didn't** have to install them, and they could be treated as private headers.

Do the contents of those files define types or classes that are actually used in the headers? Or could their `#include`s be moved to the `.cpp` files and treated as private headers, instead?

Reply:
The other reason it would be nice to make some of these headers `PRIVATE` if possible is, it's less stuff for the Python bindings to have to wrap. Anything in the installable public headers is fair game for having to be wrapped in the Python API as well, which is why I've begun working on an effort (PR #602) to migrate references to internal headers out of the other, public headers and into the corresponding `.cpp` files instead. Any header file that isn't referenced by any of the _other_ header files can be considered private and left out of both the install and the generated bindings, which will make our installed APIs tighter and leaner, plus it'll help shave a bit of time off of the builds for both our own internal code and any external API consumers.